packages/fast-components-styles-msft/src/text-field/index.ts

```diff
@@ -1,13 +1,18 @@
-import { IDesignSystem } from "../design-system";
+import designSystemDefaults, { IDesignSystem } from "../design-system";
 import { ComponentStyles, ICSSRules } from "@microsoft/fast-jss-manager";
 import { ITextFieldClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
 import { toPx } from "@microsoft/fast-jss-utilities";
+import { isEmpty } from "lodash-es";
 import { applyType } from "../utilities/typography";
 import { fontWeight } from "../utilities/fonts";
 import { applyMixedColor } from "../utilities/colors";
 
 function applyTextFieldMixedColor(mixValue?: number): (config: IDesignSystem) => string {
-    return (config: IDesignSystem): string => applyMixedColor(config.foregroundColor, config.backgroundColor, mixValue);
+    return (config: IDesignSystem): string => {
+        config = isEmpty(config) ? designSystemDefaults : config;
```

Comment:
config appears to be passed and then re-assigned, I think to be safe we would want to instead not reassign this and do something more like:

```
const colorConfig: string[] = [
    isUndefined(config.foregroundColor) ? designSystemDefaults.foregroundColor : config.foregroundColor,
    isUndefined(config.backgroundColor) ? designSystemDefaults.backgroundColor : config.backgroundColor,
    mixValue
];

return applyMixedColor(...colorConfig);
```

Reply:
I think the task may have been a bit misleading. 
```
In case there is no wrapping design system config we should have a default fallback built in.
```
First, let me explain the reasoning for addressing "config".

Config is only being reassigned if there is no design system provided (per the current task description). The reason that I'm using isEmpty() here rather than isUndefined() is that in the event of a design system provider or design system not being present, "config" comes back as an empty object. isEmpty() will check undefined, null, and an empty object and return true - which is why I chose to use it.

Based on the current task description (if no design system) it seemed more convenient and cleaner to pass defaults via config rather than pass individual params in the event there is no design system.

If there is a value missing from the design system - will there not be compile errors? Currently everything in the interface we expect is required, so if I define a partial instance, I get compile errors.

Where there are no errors is where there is no concept of a wrapping design system.

I can update to address every individual instance, I just need to know how to update the task description so that there is alignment between the ask and the final PR.

Reply:
Couple notes:

1. You could have a design system that only has one value, but not all the ones you want. eg. Here you need `foregroundColor`, what if config is just `{backgroundColor: "purple"}`. Then you will be passing undefined as the `foregroundColor` as it is missing.
2. I am uncomfortable with re-assigning config as in JavaScript when you pass an object it passes a value taht is a reference to an object, you will risk manipulating config at it's source.

Reply:
I would think that you want to check for the specific value you are looking for, and if not fallback to the default value, rather than replace the entire config which may or may not exist.

Reply:
1. Valid and accurate but I'm still not sure that you won't run into compile issues with a partial design system when it exists.
2. Fair point; expectation was that config does not exist and chose isEmpty due to it technically always being passed as an empty object.

Either way with 1 & 2, I think the confusion came down to the details in the task. I read it as a lack of a design system altogether and considering the potential compile issues due to a partial design system being invalid, I set config to a default value. 

Based on this, it seems that what we want is:

"In the event that a value passed is not defined, set a fallback."

I'll abandon the PR for now and resubmit based on that scenario.